core/ElasticSearchBulkGenerator.py

```diff
@@ -0,0 +1,155 @@
+import os
+import json
+import uuid
+from zipfile import ZipFile
+
+
+class ElasticSearchGenerator:
+    """
+    This class takes a zip file containing json files from an ontology. It iterates through those files and generates
+    "json" files that can then be used in the bulk upload of elastic search. Those files will be added to the zipfile.
+    """
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def __get_contextualized_termcode_hash(context_node, termcode_node, namespace_uuid_str):
+        context_system = context_node['system']
+        context_code = context_node['code']
+        if 'version' in context_node:
+            context_version = context_node['version']
+        else:
+            context_version = ''
+
+        termcode_system = termcode_node['system']
+        termcode_code = termcode_node['code']
+        if 'version' in termcode_node:
+            termcode_version = termcode_node['version']
+        else:
+            termcode_version = ''
+
+        context_termcode_hash_input = context_system + \
+            context_code + \
+            context_version + \
+            termcode_system + \
+            termcode_code + \
+            termcode_version
+
+        namespace_uuid = uuid.UUID(namespace_uuid_str)
+        return str(uuid.uuid3(namespace_uuid, context_termcode_hash_input))
+
+    @staticmethod
+    def __iterate_tree(node, target, parent, namespace_uuid_str):
+        # For now, we agreed upon just using the first termcode if there are more than one
+        term_code = node['termCodes'][0]
+        obj = {}
```

Comment:
@michael-82  i would suggest initialising the object directly with all the fields which you are currently setting individually after